### PR TITLE
 refactored: whitelisting and test cases

### PR DESCRIPTION
**Pull request by shubhamgupta14cse**

### CODE REVIEW

File: VWOSdk/Core/VWOClient.cs [hunk 1]
```diff
@@ -430,35 +432,59 @@ private Variation FindTargetedVariation(BucketedCampaign campaign, string campai
                     variationTargettingVariable = new Dictionary<string, dynamic>();
                     variationTargettingVariable.Add("_vwo_user_id", userId);
                 }
-                List<Variation> whiteListedVariations = this.GetWhiteListedVariationsList(userId, campaign, campaignKey, customVariables, variationTargettingVariable);
+                else
+                {
+                    if (variationTargettingVariable.ContainsKey("_vwo_user_id"))
```
Comment (softvar) [inline VWOSdk/Core/VWOClient.cs]: should be `_vwoUserId `

**Comment by softvar** [comment]
Test cases not passing for any versions?

**Comment by softvar** [comment]
I couldn't find any tests related to this https://github.com/wingify/vwo-python-sdk/blob/master/tests/data/segmentor_test_cases.json#L7379 and below it

File: VWOSdk/Core/VWOClient.cs (hunk 1 above)
Comment (shubhamgupta14cse) [reply VWOSdk/Core/VWOClient.cs]: sure , will do.

**Pull request merged**